src/commands/referral/cleanupExipredCode.ts

```diff
@@ -0,0 +1,38 @@
+import { ReferralCode } from '@prisma/client';
+import { getPrismaClient } from '../../clients';
+import { parseDate } from './parseDate';
+
+export const cleanupExipredCode = async (
+  referrals: ReferralCode[]
+): Promise<ReferralCode[]> => {
+  const prisma = getPrismaClient();
+
+  const [expiredIds, filteredReferrals] = referrals.reduce<
+    [string[], ReferralCode[]]
+  >(
+    (total, referral) => {
+      const [parseDateCase] = parseDate(referral.expiry_date.toString());
+      if (parseDateCase !== 'SUCCESS') {
+        total[0].push(referral.id);
+      } else {
+        total[1].push(referral);
+      }
+
+      return total;
+    },
+    [[], []]
+  );
+
+  // SOMETIMES try to clean expired referral codes
+  try {
+    if (expiredIds.length > 10) {
```

Comment:
cái if này nên vòng ngoài cái try/catch chứ nhỉ? Vì khi đúng condition, cái prisma call mới chạy + bắt đầu remove stuff, để actually có error => so it got caught.

a bit pedantic, but it makes more sense to me.

```ts
if (expriedIs.length > 10) {
  try {
      await prisma.referralCode.deleteMany({
        where: { id: { in: expiredIds } },
      });
  } catch (error) {
    console.error(error);
  }
}
```

Reply:
either way actually :))

Reply:
yeah ý là code nó vẫn work, mà anh nghĩ làm kiểu kia thì dễ hiểu hơn. Not too fussed.

Reply:
yeah yeah e fix roi :p